ext/RMagick/rmimage.c

```diff
@@ -635,6 +635,7 @@ Image_alpha_eq(VALUE self, VALUE type)
 {
     VALUE argv[1];
     argv[0] = type;
+    rb_warning("Image#alpha= is deprecated; use Image#alpha.");
```

Comment:
Interesting, are these methods destructive? If so, I kind of think we might want to keep them and replace the other method. I think it's more ruby-ish to have setter methods for things like this. `image.alpha = 5` vs `image.alpha(5)`. We might want to look into it in the wider context of RMagick, though.

Reply:
Found this in the comments of `Image_alpha`:

```c
/* Notes:
 *   - Replaces matte=, alpha=
 *   - Originally there was an alpha attribute getter and setter. These are
 *     replaced with alpha? and alpha(type). We still define (but don't
 *     document) alpha=. For backward compatibility, if this method is called
 *     without an argument, make it act like the old alpha getter and return
 *     true if the matte channel is active, false otherwise.
 */
```

I also think this is a better design because the "setter" will modify the image.

Reply:
Okay, so do we want to move back and have:

- `Image#alpha=` sets alpha
- `Image#alpha` returns existing alpha value

Do we need `Image#alpha?` at all? Not sure if this method is useful.

Reply:
That method returns true or false depending on the fact that the image has an alpha channel. The other will disable or enable (with various options) the alpha channel. The latter uses an enum and the first returns a boolean. It does not make sense that a property has different types for the getter and setter.

Reply:
I see. I agree it doesn't make sense for a getter to return a boolean when the setter takes an arbitrary argument. It looks like `SetImageAlphaChannel` can have a wide variety of side effects on the image, and the actual `alpha_type` is not preserved. Still, the `alpha` method feels like it's a setter, so I guess I'd argue we should have a setter and a predicate method:

* `Image#alpha=` sets alpha
* `Image#alpha?` returns `true` or `false` based on whether an alpha channel is set

Reply:
Your thoughts on this @Watson1978?

Reply:
In short, do we keep the `alpha=` method, or do we replace it with `alpha()` as the setter method. Because we can't get back the value that was set, we can't have an `alpha` getter method. Instead we can use the `alpha?` predicate method to know whether alpha was set.